src/encoded/search/lucene_builder.py

```diff
@@ -816,14 +821,126 @@ def fix_nested_aggregations(search, es_mapping):
+        is_numerical_field = field_schema and field_schema['type'] in ("integer", "float", "number")
+        if is_date_field:
+            facet['field_type'] = 'date'
+        elif is_numerical_field:
+            facet["field_type"] = field_schema['type'] or "number"
+            if "number_step" not in facet:
+                if "number_step" in field_schema:
+                    facet["number_step"] = field_schema['number_step']
+                elif facet["field_type"] == "integer":
+                    facet["number_step"] = 1
+                else:  # Default
+                    facet["number_step"] = "any"
+        facet_filters = cls.generate_filters_for_terms_agg_from_search_filters(query_field, search_filters,
+                                                                               string_query)
+        # stats aggregations could be nested too
+        stats_agg = {
+            STATS: {
+                'field': query_field
+            }
+        }
+        if nested_path:
+            facet['aggregation_type'] = 'nested:stats'
+            aggs[facet['aggregation_type'] + ':' + agg_name] = {
+                AGGS: {
+                    'primary_agg': cls._build_nested_aggregation(stats_agg, nested_path)
+                },
+                FILTER: facet_filters
+            }
+
+        else:
+            aggs[facet['aggregation_type'] + ":" + agg_name] = {
+                AGGS: {
+                    'primary_agg': stats_agg
+                },
+                FILTER: facet_filters
+            }
+
+    @classmethod
+    def _build_range_aggregation(cls, facet, query_field, search_filters, string_query, nested_path, aggs, agg_name):
+        """ Helper function for build_facets that builds range aggregations """
+        facet_filters = cls.generate_filters_for_terms_agg_from_search_filters(query_field, search_filters,
+                                                                               string_query)
+        ranges = [{k: v for k, v in r.items() if k in ['from', 'to']} for r in facet['ranges']]
+        range_agg = {
+            RANGE: {
+                FIELD: query_field,
+                'ranges': ranges
+            }
+        }
+        if nested_path:
+            facet['aggregation_type'] = 'nested:range'
+            nested_bucket_range_field = facet['aggregation_type'] + ':' + agg_name
+            aggs[nested_bucket_range_field] = {
+                AGGS: {
+                    'primary_agg': cls._build_nested_aggregation(range_agg, nested_path)
+                },
+                FILTER: facet_filters
+            }
+
+        else:
+            facet['aggregation_type'] = RANGE
+            range_field = facet['aggregation_type'] + ':' + agg_name
+            aggs[range_field] = {
+                AGGS: {
+                    'primary_agg': range_agg
+                },
+                FILTER: facet_filters
+            }
```

Comment:
This seems like it doesn't share the common parts, so I tried rearranging it (untested):

```suggestion
        if nested_path:
            facet['aggregation_type'] = 'nested:range'
            field = facet['aggregation_type'] + ':' + agg_name
            primary_agg = cls._build_nested_aggregation(range_agg, nested_path)
        else:  # regular (non-nested) range field
            facet['aggregation_type'] = RANGE
            field = facet['aggregation_type'] + ':' + agg_name
            primary_agg = range_agg
        aggs[field] = {
            AGGS: {
                'primary_agg': primary_agg
            },
            FILTER: facet_filters
        }
```

Reply:
I've done a similar refactor and applied it in a few places.